pkg/endpoint/endpoint.go

```diff
@@ -268,6 +268,10 @@ type Endpoint struct {
 	// You must hold Endpoint.Mutex to read or write it (but not to log with it).
 	logger unsafe.Pointer
 
+	// policyLogger is a logrus object with fields set to report an endpoints information.
+	// You must hold Endpoint.Mutex to read or write it (but not to log with it).
```

Comment:
nit on the comment here: How can you log with it without reading it exactly? I'm not sure I understand how the documented requirement here around holding Endpoint.Mutex fits with the read without holding the mutex in `getPolicyLogger()`.

Reply:
These comments need updating, as these pointers are accessed via atomic.LoadPointer/StorePointer only.